src/Microsoft.AspNetCore.SignalR.Client.TS/Formatters.ts

```diff
@@ -0,0 +1,146 @@
+﻿import { Message, MessageType } from './Message';
+
+let knownTypes = {
+    "T": MessageType.Text,
+    "B": MessageType.Binary,
+    "C": MessageType.Close,
+    "E": MessageType.Error
+};
+
+function splitAt(input: string, searchString: string, position: number): [string, number] {
+    let index = input.indexOf(searchString, position);
+    if (index < 0) {
+        return [input.substr(position), input.length];
+    }
+    let left = input.substring(position, index);
+    return [left, index + searchString.length];
+}
+
+export namespace SSE {
```

Comment:
They aren't "classes" though, the closest match in C# is a `static class`, since you never create an instance, they're just groupings of functions. In TypeScript, the idiomatic way to do that is to put functions in a namespace. I could use a `class` and just only put `static` functions on it, but that seems like trying to hard to make TypeScript look like C#... I'm open to changing if you'd prefer a class though.

I will definitely rename them though.